eclipse-theia/theia
# Pass plugin information through to languages-main

#### What it does

This PR passes some plugin info through to languages-main and establishes a map that allows one to get a handle for a specific vscode extension's language server call.

An example of this map is:
```
{
    "redhat.vscode-yaml": {
        "completion": 0,
        "outline": 1
    },
    "redhat.java": {
        "completion": 2,
        "outline": 3,
        "codeLens": 4
   }
}
```

Related issue: https://github.com/eclipse-theia/theia/issues/6151

#### How to test

I've created a sample theia extension https://github.com/JPinkney/Theia-Test-Extension that you clone into theia/packages/ and then add `"@theia/test-extension": "^.0.10.0"` to the package.json. Then add a plugin to the plugins folder such as vscode-java, vscode-xml. Inside of theia open a java and xml file (so the vscode extensions get activated). Then open up the browsers console and after 10 seconds you should see a printout of the map.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
